components/slottingGenerator/fn_composeJSONUnknownFragments.sqf

```diff
@@ -0,0 +1,47 @@
+params ["_structure"];
+
+if (isNil "_structure") exitWith {["null"]};
+
+switch (typeName _structure) do 
+{
+	case "STRING": {[str _structure]};
+	case "ARRAY": 
+	{
+		if (count _structure <= 0) exitWith {["[]"]};
+		private _sampleContent = _structure#0;
+
+		switch (typeName _sampleContent) do 
+		{
+			case "ARRAY": 
+			{
+				if ((count _sampleContent == 2) and ((typeName (_sampleContent#0)) == "STRING")) exitWith 
```

Comment:
Use `isEqualType` over manual type comparisons: 

```suggestion
				if ((count _sampleContent == 2) and {(_sampleContent#0) isEqualType ""}) exitWith 
```

Also note the use of `{}` for [short-circuit evaluation](https://community.bistudio.com/wiki/a_and_b), to ensure that we are always dereferencing an array index that is valid.